hack/generate/olm_bindata.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+filename="olm_bindata-"$1.go
+echo $filename
+
+function get_olm_manifests() {
+    mkdir olm-manifests
+    echo "downloading olm manifests for version ${1}"
+    wget -O olm-manifests/olm.yaml "https://github.com/operator-framework/operator-lifecycle-manager/releases/download/${1}/olm.yaml"
+    wget -O olm-manifests/crds.yaml "https://github.com/operator-framework/operator-lifecycle-manager/releases/download/${1}/crds.yaml"
+}
+
+function remove_olm_manifests {
+    rm -rf olm-manifests
```

Comment:
see that it still needs a shape. Note that:
- We can have Many OLM bindata versions (the GetAvailable returns an array) 
- The script gen just 1 version
- However, it will remove all 

I see a few options to make the behaviour consistent for we get it merged:

- Option A: Instead of we have `OLM_VERSION = "0.15.1"` in the Makefile. We need to have all Versions than the script will re-gen all always.

OR

- Option B: We just remove the `OLM_VERSION` informed as arg of the script and re-gen the specific one. 
 
Shows that Option B is easier to achieve here. However, I am ok with both. Feel free to choose. The goal here just avoids run the command and remove all bindata pkg and re-gen just 1. 

I hope that it clarifies.

Reply:
In the current scenario, when we add the manifests for a new version we need to generate the bindata for all of them again. We cannot have add more bindata to the existing file, nor have multiple bindata files in the same package.

So, the workflow would be as follows:

1. We download the manifests for a single version in `olm-manifests` file and create a bindata for it which we commit. For ex, lets say it is 0.15.1
2. Now we want to add more versions. We would remove the current bindata.go file, download olm manifests for both the versions(previous and the one which we want to add), create bindata out of it and commit to the repository.

So eventually, the command would look like:
`make gen-olm-bindata 0.15.1 0.16.1 0.x.y ...`

I expect that SDK would be supporting at max maybe 5 versions at a time.

Considering this, in the current PR, I have implemented for just first version. In a follow up PR, I will make modifications to include more than one version.

Reply:
IHMO we need to:
- Instead of  `make gen-olm-bindata 0.15.1 0.16.1 0.x.y ..` add the array of the versions in the shell script since it will be very hard to keep maintanied. Imagine we have 10 versions. How we also we keep these versions tracked in the SDK's Makefile ?
- The implementation here in this PR will not allow we accomplished the goal of have multi versions as described. So, we need to at least add a TODO comment in the code explaining what is missing to be done to achive the fully the goal.

Reply:
* `make gen-olm-bindata 0.15.1 0.16.1 0.x.y ..` isn't actually how this script works, I think @varshaprasad96 was using that example demonstratively.
* A TODO is fine, and also say we need to discuss how many versions we plan on supporting.